android/src/app/organicmaps/location/LocationHelper.java

```diff
@@ -369,6 +371,41 @@ public void stop()
     mActive = false;
   }
 
+  /**
+   * Resume location services when entering the foreground.
+   */
+  public void resumeLocationInForeground()
+  {
+    if (isActive())
+      return;
+    else if (LocationState.nativeGetMode() == LocationState.NOT_FOLLOW_NO_POSITION)
+    {
+      Logger.i(LOCATION_TAG, "Location updates are stopped by the user manually.");
+      return;
+    }
+    else if (ContextCompat.checkSelfPermission(mContext, ACCESS_FINE_LOCATION) != PERMISSION_GRANTED &&
+             ContextCompat.checkSelfPermission(mContext, ACCESS_COARSE_LOCATION) != PERMISSION_GRANTED)
+    {
+      Logger.i(LOCATION_TAG, "Permissions ACCESS_FINE_LOCATION and ACCESS_COARSE_LOCATION are not granted");
+      return;
+    }
+
+    Logger.i(LOCATION_TAG);
```

Comment:
Not needed, we already have verbose log in start()

Reply:
But in this case we will not know the caller.

Reply:
Removed.